### PR TITLE
Iss37 define scoot sublanguage

### PR DESCRIPTION
**Pull request by smover**

### CODE REVIEW

**Comment by smover** [comment]
The pull request add a list of features supported by the language (I did not define a syntax or a formal semantic) and a set of test cases for the supported features.

I would like to get feedback on how to improve the tests (e.g. is the structure of the test ok? Are there corner cases in the semantic that we want to test?) and then merge the current changes .

**Comment by smover** [comment]
@ftc This is the pull request we were talking about. Could you check if the structure of the test cases is adequate?

The idea is that these should be test cases in the java subset that the concrete interpreter should be able to run.

File: src/test/java/edu/colorado/plv/cuanto/javita/ClassDeclaration.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package edu.colorado.plv.cuanto.javita;
+
+/**
```
Comment (bechang) [inline src/test/java/edu/colorado/plv/cuanto/javita/ClassDeclaration.java:3]: Add why we have this?

File: src/test/scala/edu/colorado/plv/cuanto/javita/TestJavitaSemantic.scala [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  val mcString = "Method call"
+  val resString = "Result"
+
+  def runMethods[A](table : org.scalatest.prop.TableFor2[() => A, A]) : Unit = {
+    it should s"Compute correct result" in {
+      forAll (table) { (e, n) => {
+        (e()) should equal (n)
+      }}
+    }
+  }
```
Comment (bechang) [inline src/test/scala/edu/colorado/plv/cuanto/javita/TestJavitaSemantic.scala:20]: Flip the forAll and test so that you get not one big test but many small tests

File: src/test/scala/edu/colorado/plv/cuanto/javita/TestJavitaSemantic.scala [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+  behavior of "Arithmetic operators"
+  val exprTests = Table(
+    (mcString, resString),
+    (Expressions.testExpr1 _,3),
```
Comment (bechang) [inline src/test/scala/edu/colorado/plv/cuanto/javita/TestJavitaSemantic.scala:25]: Make this table be a mapping from "method keys" to values.
